packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -289,28 +291,21 @@ export function createComponent(_component, isAngleBracket, _props, renderNode,
   return component;
 }
 
-function shadowedAttrs(target, attrs) {
-  let shadowed = {};
-
-  // For backwards compatibility, set the component property
-  // if it has an attr with that name. Undefined attributes
-  // are handled on demand via the `unknownProperty` hook.
-  for (var attr in attrs) {
-    if (attr in target) {
-      // TODO: Should we issue a deprecation here?
-      // deprecate(deprecation(attr));
-      shadowed[attr] = attrs[attr];
-    }
+function takeSnapshot(attrs) {
+  let hash = {};
+
+  for (var prop in attrs) {
+    hash[prop] = getCellOrValue(attrs[prop]);
   }
 
-  return shadowed;
+  return hash;
 }
 
-function takeSnapshot(attrs) {
+export function takeLegacySnapshot(attrs) {
```

Comment:
Can you add a small comment inline explaining the difference between `takeLegacySnapshot` and `takeSnapshot`? It was initially confusing to me.